todolist/pages/index.js

```diff
@@ -48,11 +48,19 @@ function Home() {
     localStorage.setItem("tasks", JSON.stringify(updatedTasks));
   };
 
-  const handleDelete = (taskId) => {
-    const updatedTasks = tasks.filter((task) => task.id !== taskId);
+ // ...
+
+const handleDelete = (taskId) => {
+  const taskIndex = tasks.findIndex((task) => task.id === taskId);
+  if (taskIndex !== -1) {
+    const updatedTasks = [...tasks];
```

Comment:
Nice work on the efficient state update: The code efficiently updates the state by creating a copy of the tasks array and only modifying the copy before using setTasks to update the state.